admin/plugins.php

```diff
@@ -390,3 +426,82 @@ function perflab_render_plugin_card( array $plugin_data ) {
 	</div>
 	<?php
 }
+
+// WordPress AJAX action to handle the button click event.
+add_action( 'wp_ajax_perflab_install_activate_standalone_plugins', 'perflab_install_activate_standalone_plugins_callback' );
+
+/**
+ * Handles the standalone plugin install and activation via AJAX.
+ *
+ * @since n.e.x.t
+ */
+function perflab_install_activate_standalone_plugins_callback() {
+	if ( ! wp_verify_nonce( $_REQUEST['nonce'], 'perflab-install-activate-plugins' ) ) {
+		$status['errorMessage'] = esc_html__( 'Invalid nonce: Please refresh and try again.', 'performance-lab' );
+		wp_send_json_error( $status );
+	}
+
+	if ( ! function_exists( 'plugins_api' ) ) {
+		require_once ABSPATH . 'wp-admin/includes/plugin-install.php';
+	}
+	require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+	require_once ABSPATH . 'wp-admin/includes/class-wp-ajax-upgrader-skin.php';
+
+	$get_standalone_plugins_file_map = perflab_get_standalone_plugins_file_map();
+	$plugins_to_activate             = perflab_get_active_modules_with_standalone_plugins();
+	$plugins                         = get_plugins();
+	$status                          = array();
+
+	foreach ( $plugins_to_activate as $module_slug ) {
+
+		// Skip checking for already activated plugin.
+		if ( perflab_is_standalone_plugin_loaded( $module_slug ) ) {
+			continue;
+		}
+
+		$plugin_slug = basename( $module_slug );
+		$api         = perflab_query_plugin_info( $plugin_slug );
+		$plugin_slug = isset( $get_standalone_plugins_file_map[ $plugin_slug ] ) ? $get_standalone_plugins_file_map[ $plugin_slug ] : '';
+
+		if ( ! $plugin_slug ) {
+			$status['errorMessage'] = esc_html__( 'Invalid plugin.', 'performance-lab' );
+			wp_send_json_error( $status );
+		}
+
+		// Check if the plugin is already installed.
+		if ( ! isset( $plugins[ $plugin_slug ] ) ) {
+			$plugin_path = WP_PLUGIN_DIR . '/' . $plugin_slug;
+
+			// Check if the plugin file exists.
+			if ( ! file_exists( $plugin_path ) ) {
+				// Replace new Plugin_Installer_Skin with new Quiet_Upgrader_Skin when output needs to be suppressed.
+				$skin     = new WP_Ajax_Upgrader_Skin( array( 'api' => $api ) );
+				$upgrader = new Plugin_Upgrader( $skin );
+				$result   = $upgrader->install( $api['download_link'] );
+
+				if ( is_wp_error( $result ) ) {
+					$status['errorMessage'] = $result->get_error_message();
+					wp_send_json_error( $status );
+				} elseif ( is_wp_error( $skin->result ) ) {
+					$status['errorMessage'] = $skin->result->get_error_message();
+					wp_send_json_error( $status );
+				} elseif ( $skin->get_errors()->has_errors() ) {
+					$status['errorMessage'] = $skin->get_error_messages();
+					wp_send_json_error( $status );
+				}
+			}
+		}
+
+		$result = activate_plugin( $plugin_slug );
```

Comment:
See my other comment above, here you need to use the full path to the plugin main file, as documented for the `activate_plugin()` function.
```suggestion
		$result = activate_plugin( WP_PLUGIN_DIR . '/' . $plugin_basename );
```